editor/types/nonCr3.ts

```diff
@@ -30,23 +64,30 @@ export const NonCr3UploadSchema = z.object({
     .refine((value) => value.trim() !== "", {
       message: "Address is missing or invalid",
     }),
-  longitude: z.coerce
-    .number()
-    .min(ATX_BBOX.longitude.min, {
-      message: `Longitude is less than the minimum bounds (${ATX_BBOX.longitude.min})`,
-    })
-    .max(ATX_BBOX.longitude.max, {
-      message: `Longitude is greater than the maximum bounds (${ATX_BBOX.longitude.max})`,
-    }),
-  latitude: z.coerce
-    .number()
-    .min(ATX_BBOX.latitude.min, {
-      message: `Latitude is less than the maximum bounds (${ATX_BBOX.latitude.min})`,
-    })
-    .max(ATX_BBOX.latitude.max, {
-      message: `Latitude is greater than the maximum bounds (${ATX_BBOX.latitude.max})`,
-    }),
-  hour: z.coerce.number().int().min(0).max(23, "Hour must be between 0 and 23"),
+  longitude: z.preprocess(
+    (val) => (val ? Number(val) : null),
+    z.number({
+        invalid_type_error: "Longitude is required and must be a number",
+      })
+      .min(ATX_BBOX.longitude.min, {
+        message: `Longitude is less than the minimum bounds (${ATX_BBOX.longitude.min})`,
+      })
+      .max(ATX_BBOX.longitude.max, {
+        message: `Longitude is greater than the maximum bounds (${ATX_BBOX.longitude.max})`,
+      })
+  ),
+  latitude: z.preprocess(
+    (val) => (val ? Number(val) : null),
+    z.number({
+        invalid_type_error: "Latitude is required and must be a number",
+      })
+      .min(ATX_BBOX.latitude.min, {
+        message: `Latitude is less than the minimum bounds (${ATX_BBOX.latitude.min})`,
+      })
+      .max(ATX_BBOX.latitude.max, {
+        message: `Latitude is greater than the maximum bounds (${ATX_BBOX.latitude.max})`,
+      })
+  ),
```

Comment:
prior to this change, we were not properly validating if latitude or longitude were null. `z.coerce` is not our friend.